### ISSUE TITLE
SQL Driver Exception "Incorrect integer value: '' for column 'domain'"

### ISSUE THREAD

**Issue by FunkeMT** [fritzmg/contao-short-urls#6]
After installing the module correctly, the following error occurs when I'm trying to add the first short-url entry over the backend module:

```
Doctrine\DBAL\Exception\DriverException:
An exception occurred while executing 'INSERT INTO tl_short_urls (`domain`, `redirect`, `disable`, `tstamp`) VALUES ('', 'permanent', 0, 0)':

SQLSTATE[HY000]: General error: 1366 Incorrect integer value: '' for column 'domain' at row 1
```

I already tried to set the domain-name in _Site Structure -> DNS Settings -> Domain Name_.

**Dev Environment:**
Docker Desktop
Version 2.0.0.0-mac78 (28905)
web-image: dwst/contao-web:latest 

**Contao Version:**
4.4.21

**Stack Trace:**
```
Doctrine\DBAL\Exception\DriverException:
An exception occurred while executing 'INSERT INTO tl_short_urls (`domain`, `redirect`, `disable`, `tstamp`) VALUES ('', 'permanent', 0, 0)':

SQLSTATE[HY000]: General error: 1366 Incorrect integer value: '' for column 'domain' at row 1

  at vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:126
  at Doctrine\DBAL\Driver\AbstractMySQLDriver->convertException('An exception occurred while executing \'INSERT INTO tl_short_urls (`domain`, `redirect`, `disable`, `tstamp`) VALUES (\'\', \'permanent\', 0, 0)\':SQLSTATE[HY000]: General error: 1366 Incorrect integer value: \'\' for column \'domain\' at row 1', object(PDOException))
     (vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php:184)
  at Doctrine\DBAL\DBALException::wrapException(object(Driver), object(PDOException), 'An exception occurred while executing \'INSERT INTO tl_short_urls (`domain`, `redirect`, `disable`, `tstamp`) VALUES (\'\', \'permanent\', 0, 0)\':SQLSTATE[HY000]: General error: 1366 Incorrect integer value: \'\' for column \'domain\' at row 1')
     (vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php:158)
  at Doctrine\DBAL\DBALException::driverExceptionDuringQuery(object(Driver), object(PDOException), 'INSERT INTO tl_short_urls (`domain`, `redirect`, `disable`, `tstamp`) VALUES (\'\', \'permanent\', 0, 0)', array())
     (vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:943)
  at Doctrine\DBAL\Connection->executeQuery('INSERT INTO tl_short_urls (`domain`, `redirect`, `disable`, `tstamp`) VALUES (\'\', \'permanent\', 0, 0)')
     (vendor/contao/core-bundle/src/Resources/contao/library/Contao/Database/Statement.php:276)
  at Contao\Database\Statement->query()
     (vendor/contao/core-bundle/src/Resources/contao/library/Contao/Database/Statement.php:250)
  at Contao\Database\Statement->execute()
     (vendor/contao/core-bundle/src/Resources/contao/drivers/DC_Table.php:676)
  at Contao\DC_Table->create()
     (vendor/contao/core-bundle/src/Resources/contao/classes/Backend.php:617)
  at Contao\Backend->getBackendModule('shorturls', null)
     (vendor/contao/core-bundle/src/Resources/contao/controllers/BackendMain.php:132)
  at Contao\BackendMain->run()
     (vendor/contao/core-bundle/src/Controller/BackendController.php:55)
  at Contao\CoreBundle\Controller\BackendController->mainAction()
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:151)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:68)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:200)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (web/app_dev.php:64)
```

Please let me know if you need more informations.

**Comment by fritzmg** [comment]
> **Contao Version:**
> 4.4.21

You should always use the latest version of your respective Contao version, before reporting any kind of bugs.

Anyway, regarding your problem: try removing this line: https://github.com/fritzmg/contao-short-urls/blob/1.3.4/system/modules/short_urls/dca/tl_short_urls.php#L111
Then clear and warmup the Symfony Application Cache and see if that helps.

**Comment by FunkeMT** [comment]
Thank you for your fast response!

If your module just work with the current Contao version you should notice this in the readme or change the ```composer.json```.

Anyway, I changed
https://github.com/fritzmg/contao-short-urls/blob/1.3.4/system/modules/short_urls/dca/tl_short_urls.php#L111
to

```php
'default' => '0',
```

and it works like a charm.

If you want, I can create a pull request for this.

Thank you very much for your help!

**Comment by fritzmg** [comment]
> If your module just work with the current Contao version you should notice this in the readme or change the `composer.json`.

No, that was not my point. The point is, if errors happen, the first think you should do is update all packages - or at least all packages that could be involved. Which includes Contao in this case.



> Anyway, I changed
> https://github.com/fritzmg/contao-short-urls/blob/1.3.4/system/modules/short_urls/dca/tl_short_urls.php#L111
> to
> 
> ```
> 'default' => '0',
> ```
> 
> and it works like a charm.
> 
> If you want, I can create a pull request for this.

Have you tried removing the line as well?

**Comment by FunkeMT** [comment]
Yes, both are fine.

**delete**
https://github.com/fritzmg/contao-short-urls/blob/1.3.4/system/modules/short_urls/dca/tl_short_urls.php#L111

and

**change**
https://github.com/fritzmg/contao-short-urls/blob/1.3.4/system/modules/short_urls/dca/tl_short_urls.php#L111
to
```php
'default' => '0',
```

**Comment by fritzmg** [comment]
Changed and released as `1.3.5`, thank you :)
